### ISSUE TITLE
postgresql secret needs to support SSL certs

### ISSUE THREAD

**Issue by Fryguy** [ManageIQ/manageiq-pods#652]
If connecting to an external Postgres, there seems to be no way to specify SSL certs for the connection.  If possible, can we extend the secret to include certs, or perhaps another secret that holds the certs?

For example, if I want to point to an IBM Cloud database, then the parameters that you need for connecting and using IBM Databases for PostgreSQL

- Hostname/address
- Port
- Maintenance database : postgres
- Username
- Password
- SSL mode : Verify-Full
- Root/SSL/TLS certificate

All of the above is readily available once you provision IBM Databases for PostgreSQL service.
_________________
The current data section of the **PostgreSQL secret** contains 
-   dbname:
-   hostname: 
-   password: 
-   port: 
-   username: 

Support would be needed for the remaining Parameters. 

Doc Link:- https://cloud.ibm.com/docs/databases-for-postgresql?topic=databases-for-postgresql-getting-started

@miq-bot add-label enhancement

PRs:
- [ ] Operator configure the orchestrator #674 
- [ ] Orchestrator configure the workers https://github.com/ManageIQ/manageiq/pull/21005

**Comment by NickLaMuro** [comment]
So I did a little research into adding SSL cert support in our pods, and here is what I found:

For connecting to the database, we override the `config/database.yml` using the `DATABASE_URL` environment variable, configured here:

https://github.com/ManageIQ/manageiq-pods/blob/b856d2d3/images/manageiq-base/container-assets/container_env#L12

So any change necessary to support this will most likely entirely done on the `manageiq-pods` repo, with some supporting work via the operator to write the cert to a file system that the Rails process has access to (which has been done for other items from what I have been told).

The two attributes that need to be add to the `DATABASE_URL` are:

- `sslmode`
- `sslrootcert` (based on the ibm docs)

The `sslmode` is something specific to PG, and _should_ get passed along to the `pg` gem from the `DATABASE_URL` settings that are parsed or the `database.yml` (preference first given to `DATABASE_URL`, which is what we use in pods).

Specifically, the value for `sslmode` is assumed to be `verify-full` in this configuration, as defined by the PG docs:

https://www.postgresql.org/docs/10/libpq-ssl.html#LIBQ-SSL-CERTIFICATES

which is capitalized in the IBM docs when working with `pgAdmin`, but I assume that doesn't matter.

The other value, `sslrootcert`, needs to point to a file which would need to be first uploaded by the operator if one is provided.  Though, in the Postgres docs above it makes it seem like we can set environment variables `PGSSLROOTCERT` and `PGSSLCRL` in leu of `sslrootcert` and `sslcrl` respectively if we want to avoid some conditional logic when setting `DATABASE_URL`.  They still need to point to file paths, but it looks like the clients might be able to read that env var instead of having to pass that in as additional configuration.

(Note:  The documentation for postgresql.org is for `libpq-ssl`, which I am pretty sure the `pg` gem compiles against, so it should be a client that supports this)

If we fully want to avoid needing to add if statements to `manageiq-base/container-assets/container_env`, we could have a environment variable for `sslmode` as something like `MIQPGSSLMODE`, and default it to `prefer` (see PG docs above, as that is the default mode), and allow it to be configured to `verify-full` or other supported values.  Then `PGSSLROOTCERT` and `PGSSLCRL` can be set if `verify-full` is required by the user to fully finish that setup.

* * *

Some additional stackoverflow resources for how to configure `database.yml` and `DATABASE_URL` for SSL `verify-full` connection types:

- https://stackoverflow.com/questions/28734498/provide-ssl-certificate-to-postgresql-in-a-rails-app
- https://stackoverflow.com/questions/9741072/specify-ssl-for-heroku-external-mysql-database-connection
